tracing-core/src/field.rs

```diff
@@ -225,6 +225,17 @@ pub trait Visit {
 
     /// Visit a value implementing `fmt::Debug`.
     fn record_debug(&mut self, field: &Field, value: &dyn fmt::Debug);
+
+    /// Visit an Option
+    fn record_option(&mut self, field: &Field, value: Option<&(dyn Value + 'static)>)
+    where
+        Self: std::marker::Sized,
+    {
+        if let Some(inner_value) = value {
+            inner_value.record(field, self)
+        } else {
+        }
```

Comment:
Can this `else { }` be removed if we aren't going to do anything with it? If a serializer cares about the lack of a value they can always override this method and do a `match` themselves.